modules/orionode/index.js

```diff
@@ -19,60 +18,201 @@ var express = require('express'),
+    let conditional = endpoint.hasOwnProperty("ifProp");
+    if(conditional && options.configParams.get(endpoint.ifProp) || !conditional) { 
+        try {
+            var mod = require(endpoint.module);
+            if(typeof mod.router === 'function') {
+				var router = mod.router(options);
+				if(router === null) {
+					return; //endpoint does not want to take part in routing, quit
+				}
+                args.push(router);
+            } else {
+				//TODO log this properly
+				cosole.log("Endpoint did not provide the API 'router' function: "+JSON.stringify(enpoint, null, '\t'));
+				return;
+			}
+			if(endpoint.endpoint) { //last sanity check in case other options have been provided
+				options.app.use.apply(options.app, args);
+			}
+        }
+        catch(err) {
+            //TODO log this properly
+			console.log("Failed to load module: "+err.message);
+        }            
+    }
+}
+
+/**
+ * Function that loads endpoints from the configuration file
+ * @param {{?}[]} endpoints The array of endpoints
+ * @param {{?}} options The map of options
+ */
+function loadEndpoints(endpoints, options) {
+    if(Array.isArray(endpoints)) {
+        endpoints.forEach(function(endpoint) {
+			let args = []
+			if (typeof endpoint.endpoint === 'string') {
+				args.push(endpoint.endpoint);
+			}
+			if(endpoint.authenticated) {
+				args.push(options.authenticate);
+				args.push(checkAuthenticated);
+			}
+			if(endpoint.checkAccess) {
+				if(args.length < 2) {
+					args.push(null);
+					args.push(null);
+				}
+				args.push(checkAccessRights);
+			}
+			tryLoadRouter(endpoint, args, options)
+        });
+    }
+}
+
+/**
+ * Load any language servers mapped to the server config
+ * @param {{?}[]} servers The array of language server metadata
+ * @param {socketio} io The backing socket.IO library
+ * @param {{?}} options The map of server configuration options
+ * @since 18.0
+ */
+function loadLanguageServers(servers, io, options) {
+	if(typeof ls === 'string' && ls && options.configParams.get('orion.single.user')) {
+		let rootPath = path.join(__dirname, "language_servers");
+		if(!fs.existsSync(rootPath)) {
+			//TODO log this properly
+			console.log("'language_servers' folder does not exist. Stopped loading language servers.")
+			return;
+		}
+		addModulePath.addPath(rootPath);
+		if(Array.isArray(servers)) {
+			servers.forEach(function(ls) {
+				if(typeof ls.dirName !== "string") {
+					console.log("Language server metadata is missing 'dirName' property: "+JSON.stringify(ls, null, '\t'));
+					return;
+				}
+				let lsPath = path.join(rootPath, ls.dirName);
+				if(!fs.existsSync(lsPath)) {
+					console.log("Language server folder does not exist: "+ lsPath);
+					return;
+				}
+				addModulePath.addPath(lsPath);
+				try {
+					var server = require(serverName);
+					if(server && typeof server.createServer === 'function') {
+						// lsregistry.installServer(server.createServer(), { 
```

Comment:
nit: delete commented code.

Reply:
Its only commented for the time being - it is used in the new forthcoming language server code.